tox.ini

```diff
@@ -6,6 +6,7 @@ setenv =
 	PYTHONWARNDEFAULTENCODING = 1
 	SETUPTOOLS_ENFORCE_DEPRECATION = {env:SETUPTOOLS_ENFORCE_DEPRECATION:1}
 commands =
+	pyright --threads
```

Comment:
I am really sorry @Avasam, I had a change of heart at this.

It seems that when you install `pyright`, it [automatically install NodeJS](https://github.com/RobertCraigie/pyright-python?tab=readme-ov-file#how-pyright-for-python-works). That is a huge undertake: it is a separated programming language/environment and it does seem excessive to automatically add that to the contributors' machines (or mine).

In this scenario, I am reconsidering the first approach that you suggested (adding a new workflow to the CI). That might work better.

I am thinking that if we have a separated workflow file, it would not interfere with `skeleton`. That might be the best scenario.

Reply:
> I am reconsidering the first approach that you suggested (adding a new workflow to the CI). That might work better.
> I am thinking that if we have a separated workflow file, it would not interfere with skeleton. That might be the best scenario.

I am 100% for this solution and is my preference as well !

Reply:
Ok so here's a workflow file without touching `tox.ini` or `.github/workflows/main.yml`, or referencing pyright in `pyproject.toml`.

Incidentally it also won't be picked-up by the `collateral` job.